qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -468,6 +471,29 @@ private static void validateIncludeSection(IncludeSection includeSection, List<D
 		}
 	}
 
+	private void validateFormSectionParameters(Section section, JavaMemberInfo javaMemberInfo,
+			String baseTypeSignature, Template template, String projectUri, JavaTypeFilter filter,
+			List<Diagnostic> diagnostics) {
+		if (javaMemberInfo != null) {
+			JavaMethodInfo method = (JavaMethodInfo) javaMemberInfo;
+			Collection<RestParam> restParams = method.getRestParameters();
+			CollectHtmlInputNamesVisitor visitor = new CollectHtmlInputNamesVisitor();

```

Comment:
Only RestParam with FORM kind must be considred for the validation. 

You collect html input elements even if there are no validation to do.

Here a suggestion of your method:

```
	private void validateRenardeFormSectionParameters(Section section, JavaMemberInfo javaMemberInfo,
			String baseTypeSignature, Template template, String projectUri, JavaTypeFilter filter,
			List<Diagnostic> diagnostics) {
		if (javaMemberInfo == null) {
			return;
		}
		JavaMethodInfo method = (JavaMethodInfo) javaMemberInfo;

		Collection<RestParam> restParams = method.getRestParameters();
		if (restParams.isEmpty()) {
			return;
		}
		// Collect the @RestForm parameters
		List<RestParam> formParams = restParams.stream()
				.filter(p -> p.getParameterKind() == JaxRsParamKind.FORM)
				.collect(Collectors.toList());
		if (formParams.isEmpty()) {
			return;
		}

		// Collect all html input elements from the text nodes inside #form section
		CollectHtmlInputNamesVisitor visitor = new CollectHtmlInputNamesVisitor();
		section.accept(visitor);
		List<String> existingInputNames = visitor.getHtmlInputNames();

		// Validate @RestForm parameters
		for (RestParam param : formParams) {
			if (param.isRequired()) {
				if (existingInputNames == null || !existingInputNames.contains(param.getName())) {
					Range range = QutePositionUtility.selectStartTagName(section);
					Diagnostic diagnostic = createDiagnostic(range, DiagnosticSeverity.Warning,
							QuteErrorCode.MissingExpectedInput, param.getName());
					diagnostic.setData(new JavaBaseTypeOfPartData(baseTypeSignature));
					diagnostics.add(diagnostic);
				}
			}
		}
	}

```